paasta_tools/marathon_tools.py

```diff
@@ -779,10 +786,16 @@ def _namespaced_get_classic_service_information_for_nerve(name, namespace, soa_d
 
 def get_classic_services_running_here_for_nerve(soa_dir):
     classic_services = []
-    for name in get_classic_services_that_run_here():
-        for namespace in get_all_namespaces_for_service(name, soa_dir, full_name=False):
-            classic_services.append(_namespaced_get_classic_service_information_for_nerve(
-                name, namespace[0], soa_dir))
+    classic_services_that_run_here, puppet_service_dir_namespaces = get_classic_services_that_run_here()
+    for service in classic_services_that_run_here:
+        if service in puppet_service_dir_namespaces:
+            for namespace in puppet_service_dir_namespaces[service]:
+                classic_services.append(
+                    _namespaced_get_classic_service_information_for_nerve(service, namespace, soa_dir))
+        else:
+            for namespace in get_all_namespaces_for_service(service, soa_dir, full_name=False):
+                classic_services.append(
+                    _namespaced_get_classic_service_information_for_nerve(service, namespace[0], soa_dir))
```

Comment:
I'm confused, why on the puppet thing do we append the entire "namespace", but in this one we only do `namespace[0]` ?